components/gantry/multiaxis/multiaxis.go

```diff
@@ -138,11 +141,21 @@ func (g *multiAxis) MoveToPosition(ctx context.Context, positions, speeds []floa
 		}
 		idx += len(subAxNum)
 
-		err = subAx.MoveToPosition(ctx, pos, speed, extra)
-		if err != nil && !errors.Is(err, context.Canceled) {
-			return err
+		if g.moveSimultaneously {
+			fs = append(fs, func(ctx context.Context) error { return subAx.MoveToPosition(ctx, pos, speed, nil) })
+		} else {
```

Comment:
This control flow is somewhat confusing: the for loop either accumulates functions in a list or else runs the functions, and then later the list is either empty or full of functions, and if it's full we run the functions in there. The new variable, `fs` might be unused depending on the contents of the config file but is always created regardless.

What would you say to always appending the functions to the list here, and then down below, we either run the whole list in parallel, or else run them sequentially? That way, the for loop does one thing (put functions in the list), the `fs` variable is always used, and there's only 1 if statement about `moveSimultaneously` (no chance half of it gets modified in the future but someone forgets to update the other half).

```go
// Accumulate all the movements in a list.
fs := []rdkutils.SimpleFunc{}
for _, subAx := range g.subAxes {
    ...
    fs = append(fs, func(ctx) {return subAx.MoveToPosition()}
}

// Perform all the movements in the list.
if g.moveSimultaneously {
    return multierr.Combine(rdkutils.RunInParallel(ctx, fs), g.Stop(ctx, nil))
} else {
    for _, f := range fs {
        if err := f(); err != nil {
            return err
        }
    }
}
```

Reply:
tried reworking it this way and ran into issues so reverted to the original setup which works when `moveSimultaneously` is true or false